### PR TITLE
task(SDK-3795) - Fixes NPE

### PR DESCRIPTION
**Pull request by Anush-Shand**
- Adds null check before string comparison

### CODE REVIEW

File: clevertap-core/src/main/java/com/clevertap/android/sdk/inapp/InAppController.java [hunk 1]
```diff
@@ -706,7 +706,7 @@ private static void showInApp(Context context, final CTInAppNotification inAppNo
             return;
         }
 
-        boolean isHtmlType = inAppNotification.getType().equals(Constants.KEY_CUSTOM_HTML);
+        boolean isHtmlType = inAppNotification.getType() != null && inAppNotification.getType().equals(Constants.KEY_CUSTOM_HTML);
```
Comment (CTLalit) [inline clevertap-core/src/main/java/com/clevertap/android/sdk/inapp/InAppController.java:709]: inAppNotification.getType() -> extract once

**Review by CTLalit — COMMENTED** [review]
Check comments.

File: clevertap-core/src/main/java/com/clevertap/android/sdk/inapp/InAppController.java (hunk 1 above)
Comment (Anush-Shand) [reply clevertap-core/src/main/java/com/clevertap/android/sdk/inapp/InAppController.java]: Resolved. Just curious if it only improves readability or does it improve performance as well?

Comment (CTLalit) [reply clevertap-core/src/main/java/com/clevertap/android/sdk/inapp/InAppController.java]: just from a thread safety pov, had seen some problem in other parts of code.

**Review by CTLalit — APPROVED** [review]

**Pull request merged**